reana_client/api/client.py

```diff
@@ -486,15 +485,10 @@ def upload_to_server(workflow, paths, access_token):
                 symlink = True
             with open(path, 'rb') as f:
                 fname = os.path.basename(f.name)
-                workflow_root = get_workflow_root()
-                if not path.startswith(workflow_root):
-                    raise FileUploadError(
-                        'Files and directories to be uploaded '
-                        'must be under the workflow root directory.')
                 # Calculate the path that will store the file
                 # in the workflow controller, by subtracting
                 # the workflow root path from the file path
-                save_path = path.replace(workflow_root, '')
+                save_path = path.replace(os.getcwd(), '')
```

Comment:
Works well :+1:

**(1)** If we stop searching for workflow root, this will mean that people cannot be located in some nested directory when launching commands? IIRC they cannot submit easily from subdirs today either, so we can simply document this... so that people are not surprised.  Example:

```console
$ reana-client create -w myanalysis
$ reana-client upload mycode
$ # oh, I need to fix something
$ cd mycode/firstpart
$ vim myfirstpart.py
$ reana-client upload . -w myfirstpart
$ # oops, where this file did go? into root?
```

Ideally, we can behave analogously with how `git` behaves when being located in a subdirectory, so that users are not surprised... but this can wait. We can document current behaviour and amend later.


**(2)** Reading the specs only from the server will interfere with development stages of the workflow. Consider the following usage scenario:

```console
$ reana-client create -w myanalysis
$ vim reana.yaml # using mydata.csv and mycode.py
$ vim mycode.py
$ # ... working on mycode.py
$ # ... oops, I forgot to specify mybackground.csv too! let's fix it:
$ cp /mynfs/somewhere/mybackground.csv .
$ vim reana.yaml # adding mybackground.csv
$ reana-client upload
... oops, only mydata.csv was uploaded; why not background?
```

It is also OK for now if we document it; however for people using Serial will be surprised, as they will be frequently editing both the code and the workflow definition while working on an analysis. We have to document to always use `reana-client upload mynewfile.py` if need be manually. So, ideally, we should still parse the current reana.yaml file too to see about any user-visible changes... It can be done later, but the "get_workflow_root()" function might be back.

Reply:
**(1)** Exactly, I removed that function because I think it was adding too much complexity. I think it feels more natural to be able to upload files from any directory than _magically_ try to find the `reana.yaml` directory. In addition, we don't know the name of the reana specification file at this operation, so we wouldn't really know what to look for. I don't know what you think.

**(2)** IMHO it doesn't make sense to look for a _random_ `reana.yaml` file, it could be whatever. Another option would be to habilitate the `-f` option in the upload command so it can parse the inputs again. So, when doing `reana-client upload` do:
- If `-f` is present, parse inputs in provided `reana.yaml` file and upload them.
- Else, do the behavior of this PR, fetch specification from the server and use the inputs from there.

Reply:
(1) Agreed. Once we have `.reana` folder, we can do like `.git` does.

(2) It wouldn't be random; it would be the one that the user used during `create -f ...` step. (One could know it from the server-side or from `.reana` once we go for it.)

We can ticketise some of these for later...